tcms/bugs/tests/test_views.py

```diff
@@ -68,3 +82,70 @@ def test_reopen_a_closed_bug(self):
         self.assertRedirects(response, redirect_url)
         bug.refresh_from_db()
         self.assertTrue(bug.status)
+
+
+class TestNewBug(LoggedInTestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        user_should_have_perm(cls.tester, 'bugs.add_bug')
+
+        cls.url = reverse('bugs-new')
+
+        cls.summary = 'A shiny new bug!'
+        cls.product = ProductFactory()
+        cls.version = VersionFactory(product=cls.product)
+        cls.build = BuildFactory(product=cls.product)
+        cls.post_data = {
+            'summary': cls.summary,
+            'reporter': cls.tester.pk,
+            'product': cls.product.pk,
+            'version': cls.version.pk,
+            'build': cls.version.pk,
+        }
+
+    def test_get_view(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['page_title'], _('New bug'))
+        self.assertEqual(response.context['form_post_url'], reverse('bugs-new'))
+        self.assertTemplateUsed(response, 'bugs/mutable.html')
+
+    @patch.object(NewBugForm, 'populate')
+    def test_get_view_initial_values(self, mock_populate):
+        response = self.client.get(self.url)
+
+        form = response.context['form']
+
+        self.assertEqual(form.initial['reporter'].pk, self.tester.pk)
+        mock_populate.assert_called_once_with(-1)
```

Comment:
Several notes here: 

- all of the asserts here could be merged with the scenario above but doesn't matter much
- using mock works to assert that we tried to populate the product with -1, another possibility would be to check that the product queryset in the form didn't contain any objects (e.g. .count() == 0) for example b/c there are no products with ID -1

- version & build querysets in the form are dependent on Product. In the previous iteration there was an assert on them now it is gone. Bring it back.

Reply:
@atodorov I just want to confirm whether I should assert that the product queryset in the form doesn't contain any object.

Because the current behaviour is for it to contain all the products in the db.

Which intuitively makes sense, because if it were to contain nothing, the user won't have any product to select for the bug they are about to create.

If this isn't the expected behaviour, I can fix it and write in the test for that.

Reply:
> Because the current behaviour is for it to contain all the products in the db.

Right, you are correct. NewBugForm.populate() only changes the version & build fields because they depend on Product. I didn't pay attention to this fact.